lib/service/Controller/Farms/FarmController.dart

```diff
@@ -71,46 +73,50 @@ class FarmController {
   FarmController.fromFarm(FarmController farm) {
     this.strAddress = farm.strAddress;
     this.strName = farm.strName;
-    this.dAPR = farm.dAPR;
-    this.dTVL = farm.dTVL;
-    this.dStaked = farm.dStaked;
-    this.dRewards = farm.dRewards;
-    this.dStakeTokenPrice = farm.dStakeTokenPrice;
-    this.dRewardTokenPrice = farm.dRewardTokenPrice;
+    this.strAPR = farm.strAPR;
+    this.strTVL = farm.strTVL;
+    this.strStaked = farm.strStaked;
+    this.strRewards = farm.strRewards;
+    this.strStakeTokenPrice = farm.strStakeTokenPrice;
+    this.strRewardTokenPrice = farm.strRewardTokenPrice;
     this.strStakeTokenAddress = farm.strStakeTokenAddress;
     this.strRewardTokenAddress = farm.strRewardTokenAddress;
     this.strStakingModule = farm.strStakingModule;
     this.strStakedSymbol = farm.strStakedSymbol;
     this.strRewardSymbol = farm.strRewardSymbol;
     this.strStakedAlias = farm.strStakedAlias;
+    this.nStakeTokenDecimals = farm.nStakeTokenDecimals;
+    this.nRewardTokenDecimals = farm.nRewardTokenDecimals;
     this.rpcClient = farm.rpcClient;
     this.contract = farm.contract;
-    this.strCurrentBalance = farm.strCurrentBalance;
+    this.stakingInfo = farm.stakingInfo;
+    this.stakedInfo = farm.stakedInfo;
 
     String account = controller.publicAddress.value.hex;
     this.updateStakedBalance(account);
     this.updateCurrentBalance();
   }
 
   /// This function is used to get current Balance from wallet
-  /// 
+  ///
   /// @return {void}
   Future<void> updateCurrentBalance() async {
-    this.strCurrentBalance.value = await wallet.getTokenBalance(this.strStakeTokenAddress);
+    stakingInfo.value = await wallet.getTokenBalanceAsInfo(
+        this.strStakeTokenAddress, this.nStakeTokenDecimals);
   }
 
-
   /// This function is used to stake tokens on a specific farm
   /// and also update the txHash string on conroller
   ///
   /// @return {void}
   Future<void> stake() async {
     Uint8List stakingData = Uint8List.fromList([]);
     Uint8List rewardData = Uint8List.fromList([]);
-
-    BigInt stakeAmount = normalizeInput(this.dStakeBalance.value);
-    String txHash = await this.contract.stake(
-        stakeAmount, stakingData, rewardData,
+    UserInputInfo inputInfo = UserInputInfo.fromInput(
+        inputAmount: this.strStakeInput.value,
+        decimals: this.nStakeTokenDecimals);
+    BigInt amount = getMaximamAmount(this.stakingInfo.value, inputInfo);
```

Comment:
This should be spelled as `getMaximumAmount` not `getMaximamAmount`. Make sure to correct the spelling for all occurrences